Refactoring the way we try to build packages

dfmedeiros opened the pull request:
We were iterating over each one of the active stock locations to build packages and skipping those without the variant in stock.

This was triggering a lot of unnecessary queries to the database and this pull request addresses this issue.

Now we try to find just the stock locations that contain the requested variants in stock and build packages for each one of them.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://dfmedeiros-avoid-unnecessary-queries-on-coordinator.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://dfmedeiros-avoid-unnecessary-queries-on-coordinator.spree-spree-9ed724b77adec04c49c2.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

dfmedeiros committed:
Refactoring the way we try to build packages

We were iterating over each one of the active stock locations to build packages and skipping those without the variant in stock.

This was triggering a lot of unnecessary queries to database and this commit address this issue.

Now we try find just the stock locations that contain the requested variants in stock and build packages for each one of them.

Mafi88 commented:
:+1: 
@priyank-gupta What do you think?

priyank-gupta commented:
:+1:
